### PR TITLE
Insert content text input client breaking change

### PR DESCRIPTION
**Pull request by tneotia**
_Description of what this PR is changing or adding, and why:_

This adds a migration guide for the new member of `TextInputClient` added by https://github.com/flutter/flutter/pull/110052

_Issues fixed by this PR (if any):_ Fixes #ISSUE-NUMBER

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (justinmc) [inline src/release/breaking-changes/insert-content-text-input-client.md:56]: Some ideas of other things we might want to add:

  1. Maybe this is obvious, but it might help to state that if their implementation of TextInputClient doesn't need the ability to receive image data from Android virtual keyboards, then they can feel free to leave the implementation empty.

```dart
  @override
  void insertContent() {}
```
  2. State that instead of adding an empty implementation, if they don't want to be broken by changes like this in the future, they can use TextInputClient with the `with` keyword instead of `implements`.
  3. Maybe we should direct them to the implementation in TextInputClient as an example.  Like "TextInputClient calls simply calls a callback when `insertContent` is called.  See <link to code on GitHub when it's merged> for more details."

**Review by justinmc — APPROVED** [review]
LGTM from a technical standpoint.  I added some suggestions of details that could be added, but I think it's up to you.

Also just stating that this PR should wait until the framework PR is merged and then be updated with the version it landed in.

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:17]: **issue:** Tighten language. (Removed adverbs as extraneous.)
```suggestion
As of Android 7.1, IMEs (input method editors or virtual keyboards) can send
images and rich content into a text editor. This allows users to insert gifs,
stickers, or context-aware rich content into a text field.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:24]: **issue:** Fix passive voice.
```suggestion
This notifies the Dart code that the IME inserted rich content. 
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:30]: **issue:** Remove adverb.
```suggestion
If you had implemented `TextInputClient`, you must override
```

**Review by atsansone — CHANGES_REQUESTED** [review]
@tneotia : A little bit of language tweaking. LGTM after those changes.

**Comment by LongCatIsLooong** [comment]
Hi @atsansone could you take another look?

**Comment by justinmc** [comment]
And @tneotia can you update the branch with main?

**Comment by tneotia** [comment]
@justinmc done

**Comment by tneotia** [comment]
Hi, any movement with this? Do I need to mark this as ready to review? (I was under the assumption that the PR in the SDK repo needs to be merged first, but maybe this is the holdup instead?)

File: src/release/breaking-changes/index.md
Comment (sfshaza2) [inline src/release/breaking-changes/index.md:25]:
```suggestion
* [Insert content text input client]
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:4]: The description field in the yaml matter must be on one line.

```suggestion
title: Insert Content Text Input Client
description: Add a new method to the TextInputClient interface to allow Android virtual keyboards to insert rich content into Flutter TextFields. 
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:10]: Needs to be written in past tense.

```suggestion
Added an `insertContent` method to the `TextInputClient` interface to allow Android's image keyboard 
feature to insert content into a Flutter `TextField`.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:60]: I guess this answers my question above. An empty implementation has no consequences. But what does that mean, exactly? The rich content is still added?

```suggestion
If your implementation of `TextInputClient` doesn't require the ability
to receive rich content inserted from the IME, you can leave the implementation
of `insertContent` empty with no  consequences.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:76]: I can't land this with a TBD link. Do you have an actual link?

```suggestion
Alternatively, you can use a similar implementation to the default `TextInputClient`.
For more details, see the [insertContent implementation]({{site.repo.flutter}}/TBD).

In the future, you can prevent being broken by changes to an interface
by using `with TextInputClient` rather than `implements TextInputClient`.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:80]: Again, I can't land this with info on where it landed. The stable release field can be TBD for now.

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:25]:
```suggestion
When the user inserts rich content in the IME, the platform sends
a `TextInputClient.commitContent` channel message, notifying
the Dart code that the IME inserted rich content.
The channel message contains the mime type, URI, and bytedata for the inserted content in JSON form.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (sfshaza2) [inline src/release/breaking-changes/insert-content-text-input-client.md:31]: If they provide an empty implementation, what happens? The rich content is
still added, yes?

```suggestion
If you previously implemented the `TextInputClient` interface, you must override
`insertContent` to either support rich content insertion or provide an empty implementation.
```

**Review by sfshaza2 — DISMISSED** [review]
Thanks so much for this breaking change, @tneotia! I had a few minor nits and I can't land this without actual links and the "landed in" info. For the. missing API link, perhaps you could use a link to the master docs repo? The template explains this, but the link looks something like:  ` {{site.master-api}}/flutter/[link_to_relevant_page].html`.

**Comment by sfshaza2** [comment]
P.S. This PR would not hold up the PR on the flutter engine repo.

**Comment by sfshaza2** [comment]
Also, when this PR is ready to land, remove it from the Draft state.

**Comment by sfshaza2** [comment]
@tneotia, can we move this out of draft state and land? We are freezing the website by end of day today. Otherwise we will just close it and it can be reopened later.

**Comment by sfshaza2** [comment]
Closing for now.

**Comment by tneotia** [comment]
The PR on Flutter SDK has not been merged yet so your review criteria wouldn't be satisfied. Hoping that it is revisited soon, then I'll reopen this :)

**Comment by sfshaza2** [comment]
Sounds good!

**Comment by justinmc** [comment]
Reopening because the PR has landed and has been tagged with v3.8.0-1.0.pre ([merge commit](https://github.com/flutter/flutter/commit/0e22aca785afc937cda09ae0e3148419600a4f23)).  @tneotia Could you add that version into this PR where you put TBDs?

File: src/release/breaking-changes/index.md
Comment (AlexV525) [inline src/release/breaking-changes/index.md:20]:
```suggestion

* [Insert content text input client][]
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (AlexV525) [inline src/release/breaking-changes/insert-content-text-input-client.md:10]: Could you make a format operation to the document? I saw there are multiple blank lines that don't follow the regular behavior in this repo.

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (AlexV525) [inline src/release/breaking-changes/insert-content-text-input-client.md:52]: This makes it easy to copy-paste directly.

```suggestion
  // ...
  @override
  void insertContent() {
    // Implement insertContent.
  }
  // ...
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (tneotia) [reply src/release/breaking-changes/insert-content-text-input-client.md:10]: Is this some sort of command I need to run in the terminal? Sorry, not super familiar with what tools are available when creating markdown docs.

Comment (AlexV525) [reply src/release/breaking-changes/insert-content-text-input-client.md:10]: Maybe a format action from editors? Such as VS Code and IDEA, or other markdown editors should support formatting too.

**Review by AlexV525 — APPROVED** [review]

**Review by domesticmouse — APPROVED** [review]

**Comment by atsansone** [comment]
@tneotia : One more change needed: There's a conflict in your PR. Resolve that and I think we could be good.

**Comment by tneotia** [comment]
@atsansone should be good now :)

**Comment by domesticmouse** [comment]
@sfshaza2 and @atsansone can you verify that your requested changes have been appropriately landed, thanks!

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:26]: **nit:** Rephrase to be more concise.
```suggestion
If you previously the `TextInputClient` interface earlier, override
`insertContent` to either support rich content insertion or provide an empty implementation.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:55]: **suggestion:** Rephrase to make more concise.
```suggestion
Your implementation of `TextInputClient` might not require the ability to receive rich content
inserted from the IME. If that's the case, you can leave the implementation of `insertContent`
empty with no consequences.
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:68]: **issue:** Redundant copy.
```suggestion
As an alternative, you can use a similar implementation to the default `TextInputClient`.
To learn how to do this, see the [insertContent implementation]({{site.master-api}}/flutter/services/TextInputClient/insertContent.html).
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (atsansone) [inline src/release/breaking-changes/insert-content-text-input-client.md:71]: **issue:** Too much passive voice. Please revise.
```suggestion
To prevent breaking changes to an interface,
use `with TextInputClient` rather than `implements TextInputClient`.
```

**Review by atsansone — DISMISSED** [review]
Needs a few tweaks, then I can approve.

**Comment by tneotia** [comment]
@atsansone should be good to go now

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (parlough) [inline src/release/breaking-changes/insert-content-text-input-client.md:68]:
```suggestion
To learn how to do this, see the [insertContent implementation]({{site.api}}/flutter/services/TextInputClient/insertContent.html).
```

File: src/release/breaking-changes/insert-content-text-input-client.md
Comment (parlough) [inline src/release/breaking-changes/insert-content-text-input-client.md:76]: Looks like this landed in 3.10.0:

```suggestion
In stable release: 3.10.0
```

**Review by parlough — APPROVED** [review]
Thanks for making those adjustments!

**Comment by Hixie** [comment]
I had a conflict with this in my PR and I'm not sure I merged it correctly, can someone check https://github.com/flutter/website/pull/8683/files is correct?

**Comment by parlough** [comment]
Yep. I'm checking and pushing fixes for it.

**Pull request merged**
